### PR TITLE
Sampling Variability Module

### PR DESCRIPTION
**Pull request by JTPetter**
Adding an analysis for sampling variability and the options which samples to display in CLT.

### CODE REVIEW

File: R/LSTcentralLimitTheorem.R [hunk 1]
```diff
@@ -56,70 +70,137 @@ LSTcentralLimitTheorem <- function(jaspResults, dataset, options) {
...
 
-.cltTakeSamples <- function(jaspResults, options, data) {
+.cltTakeSamples <- function(jaspResults, options, data, replace = TRUE) {
   n <- options[["cltSampleSize"]]
   k <- options[["cltSampleAmount"]]
   sampleList <- list()
-  for (i in 1:k) {
-    sampleList[[i]] <- sample(x = data[["x"]], size = n, replace = TRUE)
+  if (replace) {
+    for (i in 1:k) {
```
Comment (vandenman) [inline R/LSTcentralLimitTheorem.R:84]: `1:k` -> `seq_len(k)`

File: R/LSTcentralLimitTheorem.R [hunk 2]
```diff
@@ -56,70 +70,137 @@ LSTcentralLimitTheorem <- function(jaspResults, dataset, options) {
...
+      sampleIndiceList <- lapply(sampleList, .getIndices, population = orderedData)
+    }
+    return(list("samples" = sampleList,
+                "indices" = sampleIndiceList))
+  }
+}
+
+.getIndices <- function(sampleVector, population){
+  sampleIndices <- c()
+  for (i in 1:length(sampleVector)){
```
Comment (vandenman) [inline R/LSTcentralLimitTheorem.R:118]: `1:length(...)` -> `seq_along(...)`

File: R/LSTsampleVariability.R [hunk 3]
```diff
@@ -16,279 +16,288 @@
...
+        plot <- plotList[[index]]
+      }
+      plotMat[[r, c]] <- plot
+    }
+  }
+  return(plotMat)
+}
+
+.getFromToSampleShow <- function(type = c("first", "last", "range", "all"), maxSamples, singleValue = "", start ="", stop = ""){
+  if (type == "first") {
```
Comment (vandenman) [inline R/LSTsampleVariability.R:265]: for safety, perhaps add `type <- match.arg(type)`?
```suggestion
.getFromToSampleShow <- function(type = c("first", "last", "range", "all"), maxSamples, singleValue = "", start ="", stop = ""){
  type <- match.arg(type)
  if (type == "first") {
```

**Review by vandenman — APPROVED** [review]
Looks good, some mainly stylistic comments below

File: R/LSTcentralLimitTheorem.R (hunk 1 above)
Comment (JTPetter) [reply R/LSTcentralLimitTheorem.R:84]: Ah, I will change this everywhere. Very useful, thanks!

File: R/LSTcentralLimitTheorem.R (hunk 2 above)
Comment (JTPetter) [reply R/LSTcentralLimitTheorem.R:118]: This too.

**Pull request merged**